contracts/token/airdrop/AirdropSeason1.sol

```diff
@@ -0,0 +1,111 @@
+     * @param _merkleRoot           The merkle root with deposits encoded into it as hash [address, amount]
+     * @param _expiration           The expiration of the airdrop
+     * @param _votingVault          The voting vault to deposit tokens to
+     */
+    constructor(
+        IERC20 _token,
+        bytes32 _merkleRoot,
+        uint256 _expiration,
+        IAirdropSingleSidedStaking _votingVault
+    ) ArcadeAirdropBase(_token, _merkleRoot, _expiration) {
+        if (address(_votingVault) == address(0)) revert AA_ZeroAddress("votingVault");
+
+        votingVault = _votingVault;
+    }
+
+    // ===================================== CLAIM FUNCTIONALITY =======================================
+
+    /**
+     * @notice Claims an amount of tokens in the tree and stakes the tokens in the ArcadeSingleSidedStaking
+     *         contract. The caller can choose who to delegate their voting power to by passing in the
+     *         address of their delegate. If a user already has an active stake in the ArcadeSingleSidedStaking
+     *         voting vault they must use the same delegate address they are already delegating to.
+     *
+     * @param delegate               The address the user will delegate to
+     * @param totalGrant             The total amount of tokens the user was granted
+     * @param merkleProof            The merkle proof showing the user is in the merkle tree
+     * @param lock                   The locking period to apply to the staked tokens
+     */
+    function claimAndStake(
+        address delegate,
+        uint128 totalGrant,
+        bytes32[] calldata merkleProof,
+        IAirdropSingleSidedStaking.Lock lock
+    ) external {
+        if (rewardsRoot == bytes32(0)) revert AA_NotInitialized();
+        // must be before the expiration time
+        if (block.timestamp > expiration) revert AA_ClaimingExpired();
+        // validate the withdraw
+        _validateWithdraw(totalGrant, merkleProof);
+
+        // approve the voting vault to transfer tokens
+        token.approve(address(votingVault), uint256(totalGrant));
+        // stake tokens in voting vault for this msg.sender and delegate
+        votingVault.airdropReceive(msg.sender, totalGrant, delegate, lock);
+
+        emit ClaimAndStaked(msg.sender, totalGrant, delegate, lock);
+    }
+
+    /**
+     * @notice Claims an amount of tokens in the tree and send them to the caller.
+     *
+     * @param totalGrant             The total amount of tokens the user was granted
+     * @param merkleProof            The merkle proof showing the user is in the merkle tree
+     */
+    function claim(uint128 totalGrant, bytes32[] calldata merkleProof) external nonReentrant {
+        if (rewardsRoot == bytes32(0)) revert AA_NotInitialized();
+        // must be before the expiration time
+        if (block.timestamp > expiration) revert AA_ClaimingExpired();
+        // validate the withdraw
+        _validateWithdraw(totalGrant, merkleProof);
```

Comment:
Not sure a better name, but I was a bit surprised here that `_validateWithdraw` was not `view`, in my mind "validate" in a function name implies behavior that it will check values and error, so the storage update to `claim` was a bit of a surprise

Reply:
Yeah I could pull the state change in claim out to the inherting contract, but was trying to be as least invasive with the changes since I was changing audited code. Same with the ArcadeStakingRewards contract that was how it was done originally and since it was audited was trying to lower my risk by not moving too much around. 

But what I am gathering from these comments is I should take it one step further and make it as clean as possible by making validations view and have a combined `ArcadeAirdropBase` contract. Will make these changes.